### PR TITLE
Handle malformed replies in delivery processor

### PR DESCRIPTION
**Pull request by yolocs**
Related to https://github.com/google/knative-gcp/issues/1362
Hopefully this could provide some insight for some of the test flakiness.

## Proposed Changes

- Copied workaround from https://github.com/knative/eventing/pull/3450
-
-

**Release Note**

```release-note
Malformed replies will be treated as error (instead of being silently ignored).
```

**Docs**

### CODE REVIEW

File: pkg/broker/handler/processors/deliver/processor_test.go [hunk 1]
```diff
@@ -295,38 +324,10 @@ func TestDeliverFailure(t *testing.T) {
...
-					return
-				}
-
-				// Due to https://github.com/cloudevents/sdk-go/issues/433
-				// it's not possible to use Receive to easily return error.
-				if err := resp(rctx, nil, &cehttp.Result{StatusCode: http.StatusInternalServerError}); err != nil {
-					t.Errorf("unexpected error from target responding event: %v", err)
-				}
-			}()
-
```
Comment (grac3gao) [inline pkg/broker/handler/processors/deliver/processor_test.go:324]: Just curious, why we delete this part of test?

Comment (yolocs) [reply pkg/broker/handler/processors/deliver/processor_test.go:324]: This code was used to respond events. Now it's replaced by my own handler which is needed in order to reply specific response body.

**Pull request merged**